gitlab/client.py

```diff
@@ -698,7 +712,11 @@ def http_request(
                 "retry_transient_errors", self.retry_transient_errors
             )
             if (429 == result.status_code and obey_rate_limit) or (
-                result.status_code in [500, 502, 503, 504] and retry_transient_errors
+                (
+                    result.status_code in [500, 502, 503, 504]
+                    or 520 <= result.status_code <= 530
+                )
```

Comment:
As this list grows I'm thinking we might get more arbitrary codes in this list - would it make sense to put it in a helper or constant? Then we can easily document/comment where these status codes come from.

Reply:
I've added `RETRYABLE_TRANSIENT_ERROR_CODES`